drivers/i2s/i2s_ll_stm32.c

```diff
@@ -1117,37 +1185,44 @@ struct queue_item rx_5_ring_buf[CONFIG_I2S_STM32_RX_BLOCK_COUNT + 1];
 struct queue_item tx_5_ring_buf[CONFIG_I2S_STM32_TX_BLOCK_COUNT + 1];
 
 static struct i2s_stm32_data i2s_stm32_data_5 = {
-	.dma_name = I2S5_DMA_NAME,
 	.rx = {
-		.dma_channel = I2S5_DMA_CHAN_RX,
+		.dma_name = DT_I2S_5_DMA_CONTROLLER_RX,
+		.dma_channel = DT_I2S_5_DMA_CHANNEL_RX,
 		.dma_cfg = {
 			.block_count = 1,
-			.dma_slot = I2S5_DMA_SLOT_RX,
-			.channel_direction = PERIPHERAL_TO_MEMORY,
-			.source_data_size = 1,  /* 16bit default */
-			.dest_data_size = 1,    /* 16bit default */
-			.source_burst_length = 0, /* SINGLE transfer */
-			.dest_burst_length = 1,
+			.dma_slot = DT_I2S_5_DMA_SLOT_RX,
+			.source_data_size = DT_I2S_5_DMA_SRC_DATA_SIZE_RX,  /* 16bit default */
+			.dest_data_size = DT_I2S_5_DMA_DST_DATA_SIZE_RX,    /* 16bit default */
+			.source_burst_length = DT_I2S_5_DMA_SRC_BURST_LENGTH_RX, /* SINGLE transfer */
+			.dest_burst_length = DT_I2S_5_DMA_DST_BURST_LENGTH_RX,
+			.channel_priority = DT_I2S_5_DMA_PRIORITY_RX,
```

Comment:
I had to add all of this stuff in soc/arm/st_stm32/stm32f1/dts_fixup.h
#define DT_I2S_5_DMA_CONTROLLER_TX          DT_ST_STM32_I2S_40015000_DMA_CONTROLLER_0
#define DT_I2S_5_DMA_CHANNEL_TX             DT_ST_STM32_I2S_40015000_DMA_CHANNEL_0
#define DT_I2S_5_DMA_SLOT_TX                DT_ST_STM32_I2S_40015000_DMA_SLOT_0
#define DT_I2S_5_DMA_SRC_DATA_SIZE_TX       DT_ST_STM32_I2S_40015000_DMA_SRC_DATA_SIZE_0
#define DT_I2S_5_DMA_DST_DATA_SIZE_TX       DT_ST_STM32_I2S_40015000_DMA_DST_DATA_SIZE_0
[...]

I did it only for DT_I2S_5 just to test my board. We need to add also the other stuff for all I2S controllers for all
variants. 

@erwango 
Do we really need to add this in dts_fixup or can we use in drivers the correct DTS generated names?

Reply:
This as far as I know actually should be put in dts_fixup.h. I didn't add this because I thought there was only F4 support for previous DMA and I only added 5 I2S dts fixup defines to soc/arm/st_stm32/stm32f4/dts_fixup.h. Since other platforms need this, too, I'll add it later. 
Driver cannot use them directly because the register base address changes from device to device, in this circumstance it's 40015000, and for another it becomes 40016000 or something else.

Reply:
> Driver cannot use them directly because the register base address changes from device to device, in this circumstance it's 40015000, and for another it becomes 40016000 or something else.

Correct.

Reply:
For example, in case of LPS22HB sensor the dts scripts generate following macros:

#define DT_ST_STM32_I2C_V1_40005800_ST_LPS22HB_PRESS_5D_LABEL    "LPS22HB"
#define DT_ST_LPS22HB_PRESS_0_LABEL        DT_ST_STM32_I2C_V1_40005800_ST_LPS22HB_PRESS_5D_LABEL

And in the driver we no longer need to use dts_fixup, but just DT_ST_LPS22HB_PRESS_0_LABEL.

Here, maybe, we can still use dts_fixup file and let @erwango comment on this.

Reply:
Sure, but I suspect Erwan is a bit busy these days, don't get too many responses from him, I can tell you what I'm thinking.
This is actually because you have only one LPS22HB sensor. The problem is if we enabled only one I2S sensor, no matter which one it is, dts is going to generate it as DT_ST_STM32_..._0_LABEL. This means it can represent I2S1, I2S2 or any one of them.
Like if we enabled I2S2 and I2S3, it will generate DT_ST_STM32_I2S_0_LABEL and DT_ST_STM32_I2S_1_LABEL to represent I2S2 and I2S3, so our driver would not be able to identify which device instance we are actually using. This depends on the logic of extract/reg.py. So base address like 40005800 and 40015000 and dts_fixups are always needed in our case, at least for the stm32 drivers I've been experienced.

Reply:
> And in the driver we no longer need to use dts_fixup, but just DT_ST_LPS22HB_PRESS_0_LABEL.

This is not clear yet how we'll use "indexed instances" (a continuous index starting from 0, generated at compilation depending on the enabled IP instances. Ex UART_1 UART_3 UART_5 will get _0, _1 and _2). Basically this works today where there is only one IP instance, sensors being an example.

So for now, dts_fixup is the way.